src/zope/app/broken/broken.py

```diff
@@ -24,33 +23,32 @@
 from zope.broken.interfaces import IBroken
 from zope.annotation.interfaces import IAnnotations
 
+@zope.interface.implementer(
+    IBroken,
+    zope.location.interfaces.ILocation,
+    IAnnotations,
+)
 class Broken(ZODB.broken.Broken):
-    zope.interface.implements(
-        IBroken,
-        zope.location.interfaces.ILocation,
-        IAnnotations,
-        )
 
-    def __parent__(self):
-        return self.__Broken_state__.get('__parent__')
+    def __get_state(self, key, default=None):
+        get = getattr(self.__Broken_state__, 'get', None)
+        if get is None:
+            get = lambda k, d: default
+        return get(key, default)
 
-    __parent__ = property(__parent__)
+    @property
+    def __parent__(self):
+        return self.__get_state('__parent__')
```

Comment:
It seems that this now handles more brokenness than before.  This may be worth mentioning in the changelog.